File: Frontend/src/api/payee.ts
import api from './client';

export type PayeeListItemDto = {
  id:               number;
  initials:         string;
  avatarColor:      string;
  payeeName:        string;
  payeeType:        string;
  claimId:          string;
  policyId:         string;
  paymentMethod:    string;
  complianceCheck:  string;
  dateOfLoss:       string;
  lossType:         string;
  amount:           string;
  status:           string;
};

export type PayeeStatusKpiDto = {
  total:    number;
  draft:    number;
  pending:  number;
  approved: number;
  rejected: number;
};

export type CreateAddressRequest = {
  googleAddress: string | null;
  isManual:      boolean;
  addressLine1:  string | null;
  addressLine2:  string | null;
  country:       string | null;
  state:         string | null;
  city:          string | null;
  county:        string | null;
  zipCode:       string | null;
  latitude:      string | null;
  longitude:     string | null;
};

export type CreatePayeeRequest = {
  claimId:              number | null;
  claimPayeeType:       string | null;
  firstName:            string | null;
  middleName:           string | null;
  lastName:             string | null;
  businessName:         string | null;
  nationalId:           string | null;
  socialSecurityNumber: string | null;
  tinId:                string | null;
  payeeType:            string | null;
  relationship:         string | null;
  telephoneNumberCC:    string | null;
  telephoneNumber:      string | null;
  email:                string | null;
  bankDetail: {
    paymentMethod:               string | null;
    bankName:                    string | null;
    accountHolderName:           string | null;
    accountNumber:               string | null;
    accountType:                 string | null;
    abaRoutingNumber:            string | null;
    businessName:                string | null;
    federalTaxClassification:    string | null;
    employerIdentificationNumber: string | null;
    type1099:                    string | null;
    w9FormOnFile:                boolean;
  } | null;
  address: CreateAddressRequest | null;
};

export type CreateBankDetailRequest = {
  paymentMethod:               string | null;
  bankName:                    string | null;
  accountHolderName:           string | null;
  accountNumber:               string | null;
  accountType:                 string | null;
  abaRoutingNumber:            string | null;
  businessName:                string | null;
  federalTaxClassification:    string | null;
  employerIdentificationNumber: string | null;
  type1099:                    string | null;
  w9FormOnFile:                boolean;
};

export const payeeApi = {
  getListByClaim: (claimId: number) =>
    api.get<PayeeListItemDto[]>(`/payees/by-claim/${claimId}`).then(r => r.data),

  getKpiByClaim: (claimId: number) =>
    api.get<PayeeStatusKpiDto>(`/payees/kpi/by-claim/${claimId}`).then(r => r.data),

  create: (req: CreatePayeeRequest) =>
    api.post<{ id: number }>('/payees', req).then(r => r.data),

  updateBanking: (id: number, req: CreateBankDetailRequest) =>
    api.put(`/payees/${id}/banking`, req).then(r => r.data),

  sendForApproval: (id: number) =>
    api.post(`/payees/${id}/send-for-approval`).then(r => r.data),
};

File: Backend/src/InsureEdge.Domain/Entities/Payee.cs
namespace InsureEdge.Domain.Entities;

public class Payee
{
    public long Id { get; set; }
    public long? ClaimId { get; set; }
    public long ClientId { get; set; }

    public string? FirstName { get; set; }
    public string? MiddleName { get; set; }
    public string? LastName { get; set; }
    public string? BusinessName { get; set; }
    public string? Relationship { get; set; }
    public string? TelephoneNumberCC { get; set; }
    public string? TelephoneNumber { get; set; }
    public string? Email { get; set; }
    public string? NationalId { get; set; }
    public string? SocialSecurityNumber { get; set; }
    public string? TinId { get; set; }
    public string? PayeeType { get; set; }
    public string? ClaimPayeeType { get; set; }
    public string? PayeeCode { get; set; }

    // Compliance / checks
    public string? OFACChecks { get; set; }
    public string? BankDetailsValidation { get; set; }
    public string? DuplicatePayeeCheck { get; set; }
    public string? FraudCheck { get; set; }
    public string? CertificationVerification { get; set; }

    public string Status { get; set; } = "Draft";
    public string? Comments { get; set; }

    public long? ApprovedBy { get; set; }
    public DateTimeOffset? ApprovedOn { get; set; }
    public long? CreatedBy { get; set; }
    public DateTimeOffset CreatedOn { get; set; } = DateTimeOffset.UtcNow;
    public long? UpdatedBy { get; set; }
    public DateTimeOffset? UpdatedOn { get; set; }

    // Navigation
    public Claim? Claim { get; set; }
    public ICollection<BankDetail>    BankDetails { get; set; } = [];
    public ICollection<CommonAddress> Addresses   { get; set; } = [];
}
